### PR TITLE
refactor: refine greptimedb-operator e2e

### PR DESCRIPTION
**Pull request by zyy17**
## What's changed

1. Add `tests/e2e/setup/*.sh`

   - Deploy a completed Kubernetes by using kind;
   
   - Build and deploy greptimedb-operator instead of starting by calling manager function;

2. Simplify greptimedbcluster_test.go:

   - Use the Kubernetes cluster directly and load test cases from `tests/e2e/testdata/*`;
   - Use [cloud-kind-provider](https://github.com/kubernetes-sigs/cloud-provider-kind) to use LoadBalancer service in kind and the e2e can connect the greptimedb cluster from the host;

5. Add `README.md` to introduce the E2E;

6. Add remote-wal e2e;
7. Add greptimedbstandalone e2e;

### CODE REVIEW

**Comment by daviderli614** [comment]
[docker/setup-qemu-action](https://github.com/GreptimeTeam/greptimedb-operator/blob/fad3cdd004392dd3433af5cc4163d7dc8a0d0a28/.github/workflows/release.yaml#L34) Bump version to v3，[docs](https://github.com/docker/setup-qemu-action). 

[docker/setup-buildx-action](https://github.com/GreptimeTeam/greptimedb-operator/blob/fad3cdd004392dd3433af5cc4163d7dc8a0d0a28/.github/workflows/release.yaml#L37) Bump version to v3，[docs](https://github.com/docker/setup-buildx-action). 

 [docker/login-action](https://github.com/GreptimeTeam/greptimedb-operator/blob/fad3cdd004392dd3433af5cc4163d7dc8a0d0a28/.github/workflows/release.yaml#L46) Bump version to v3，[docs](https://github.com/docker/login-action). 

[docker/build-push-action](https://github.com/GreptimeTeam/greptimedb-operator/blob/fad3cdd004392dd3433af5cc4163d7dc8a0d0a28/.github/workflows/release.yaml#L46)  and [here](https://github.com/GreptimeTeam/greptimedb-operator/blob/fad3cdd004392dd3433af5cc4163d7dc8a0d0a28/.github/workflows/release.yaml#L71) Bump version to v5，[docs](https://github.com/docker/build-push-action).

File: tests/e2e/setup/create-cluster.sh [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+  # Wait for etcd to be ready.
+  kubectl wait \
+    --for=condition=Ready \
+    pod -l app.kubernetes.io/instance=etcd \
+    -n "$ETCD_NAMESPACE" \
+    --timeout="$DEFAULT_TIMEOUT"
+
+  # Wait for kafka to be ready.
+  check_kafka_cluster_status
+
```
Comment (daviderli614) [inline tests/e2e/setup/create-cluster.sh:251]: suggestion: add wait for greptimedb ready.

Comment (zyy17) [reply tests/e2e/setup/create-cluster.sh:251]: We don't deploy greptimedb in `create-cluster.sh`.

Comment (daviderli614) [reply tests/e2e/setup/create-cluster.sh:251]: Ignore it.

**Review by daviderli614 — APPROVED** [review]
LGTM

**Pull request merged**